specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2023-07-01-preview/ResourceHealth.json

```diff
@@ -0,0 +1,2167 @@
+            },
+            "lastUpdateTime": {
+              "type": "string",
+              "description": "It provides the Timestamp for when the health impacting event was last updated.",
+              "format": "date-time"
+            },
+            "hirStage": {
+              "type": "string",
+              "description": "Stage for HIR Document"
+            },
+            "additionalInformation": {
+              "type": "object",
+              "description": "Additional information",
+              "properties": {
+                "message": {
+                  "type": "string",
+                  "description": "Additional information Message"
+                }
+              }
+            },
+            "duration": {
+              "type": "integer",
+              "description": "duration in seconds",
+              "format": "int32"
+            },
+            "impactType": {
+              "type": "string",
+              "description": "The type of the impact"
+            },
+            "maintenanceId": {
+              "type": "string",
+              "description": "Unique Id for Planned maintenance event"
+            },
+            "maintenanceType": {
+              "type": "string",
+              "description": "The type of Planned maintenance event"
+            },
+            "argQuery": {
+              "type": "string",
+              "description": "ARG Query to fetch the affected resources from their existing ARG locations"
+            }
+          }
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "link": {
+      "type": "object",
+      "description": "Useful links for service health event.",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Type of link.",
+          "enum": [
+            "Button",
+            "Hyperlink"
+          ],
+          "x-ms-enum": {
+            "name": "LinkTypeValues",
+            "modelAsString": true
+          }
+        },
+        "displayText": {
+          "type": "object",
+          "description": "Display text of link.",
+          "properties": {
+            "value": {
+              "type": "string",
+              "description": "Display text of link."
+            },
+            "localizedValue": {
+              "type": "string",
+              "description": "Localized display text of link."
+            }
+          }
+        },
+        "extensionName": {
+          "type": "string",
+          "description": "It provides the name of portal extension to produce link for given service health event."
+        },
+        "bladeName": {
+          "type": "string",
+          "description": "It provides the name of portal extension blade to produce link for given service health event."
+        },
+        "parameters": {
```

Comment:
this object definition doesn't include a proper type declaration that explains how the mapping would be used by an API consumer.